File: docs/ergonomics/scripts/matlab/model_eval.m
function model_eval(path, out)
%MODEL_EVAL  Evaluate a Python-written callable file in MATLAB, exactly
%   as matlab/README.md spells it, and check 1.45 and the six pressures.

d = mestra.read(path);
t = table(0.5, 4.0, 'VariableNames', {'mach', 'alpha'});
e = mestra.evaluate(d, t);

cl = e.scalar('cl').values;
fprintf('    cl = %.17g\n', cl);

a = e.nodeArray('s0', 'pressure');
fprintf('    pressure dims: %s\n', strjoin(a.dims, ', '));
p = mestra.permute(a.values, a.dims, {'row', 'node', 'component'});
fprintf('    pressure =');
fprintf(' %g', p(1, :, 1));
fprintf('\n');

fprintf('    support id: %s\n', d.support('s0').supportId(1:16));
mestra.write(e, out);
fprintf('    wrote %s\n', out);
end

File: docs/ergonomics/scripts/julia/model_eval.jl
# Evaluate a Python-written callable file in Julia, as julia/README.md
# spells it.  Run: julia --project=julia model_eval.jl MODEL.mes OUT.mes

using Mestra

path, out = ARGS[1], ARGS[2]
ds = Mestra.read(path)
e = Mestra.evaluate(ds, Dict("mach" => [0.5], "alpha" => [4.0]))

println("    cl = ", Mestra.values(e, e.scalars["cl"]))
p = Mestra.values(e, e["pressure"])
println("    pressure dims: ", Mestra.dimnames(p))
q = Mestra.permute(p, (:row, :node, :component))
println("    pressure = ", [q[1, n, 1] for n in 1:6])
println("    support id: ", ds.supports[1].support_id[1:16])

Mestra.write(e, out)
println("    wrote ", out)
